### PR TITLE
AI scanning, new naming strategy implemented and other improvements

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

File: src/EShopWorld.Tools/Commands/AzScan/AzScanAppInsightsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        public AzScanAppInsightsCommand(Azure.IAuthenticated authenticated, KeyVaultClient keyVaultClient, IBigBrother bigBrother, ApplicationInsightsManagementClient azClient) : base(authenticated, keyVaultClient, bigBrother)
+        {
+            _azClient = azClient;
+        }
+
+        protected override async Task<int> RunScanAsync(IAzure client)
+        {
+            _azClient.SubscriptionId = client.SubscriptionId;
+
+            string nextPageLink = string.Empty;
```
Comment (arturek) [inline src/EShopWorld.Tools/Commands/AzScan/AzScanAppInsightsCommand.cs]: assignment is not necessary. Maybe while(true) {... if(ais.NextPageLink == null) break; } - but the current version is perfectly good.

Comment (arturek) [reply src/EShopWorld.Tools/Commands/AzScan/AzScanAppInsightsCommand.cs]: You're right. I haven't noticed that part.

**Comment by VaclavK** [comment]
@arturek thanks for the review

File: src/EShopWorld.Tools/Commands/AzScan/AzScanAppInsightsCommand.cs (hunk 1 above)
Comment (arturek) [reply src/EShopWorld.Tools/Commands/AzScan/AzScanAppInsightsCommand.cs]: Still I would probably move var ais = [...]ListAsync|ListByResourceGroupAsync[...] outside the loop. It should make it simpler.

File: src/Tests/EShopWorld.Tools.Tests/StringExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using EShopWorld.Tools.Commands.AzScan;
+using FluentAssertions;
+using EShopWorld.Tools.Helpers;
+using Xunit;
+
+namespace EshopWorld.Tools.Tests
+{
+    public class StringExtensionTests
+    {
+        [Theory, IsUnit]
```
Comment (dguerin) [inline src/Tests/EShopWorld.Tools.Tests/StringExtensionsTests.cs]: Beautiful, love theories.

File: src/EShopWorld.Tools/Commands/AzScan/AzCosmosDbScanCommand.cs [hunk 3]
```diff
@@ -21,14 +21,14 @@ protected override async Task<int> RunScanAsync(IAzure client)
...
-                    continue;
+                //no region filter for cosmos
 
                 var keys = await cosmos.ListKeysAsync();
                 var name = cosmos.Name.Contains('-')
                     ? cosmos.Name.Remove(cosmos.Name.LastIndexOf('-')) : cosmos.Name;
 
-                await SetKeyVaultSecretAsync(name, keys.PrimaryMasterKey);
+                await SetKeyVaultSecretAsync("CosmosDB", name, "PrimaryConnectionString", $"AccountEndpoint={cosmos.DocumentEndpoint};AccountKey={keys.PrimaryMasterKey}");
+                await SetKeyVaultSecretAsync("CosmosDB", name, "PrimaryMasterKey", keys.PrimaryMasterKey);
```
Comment (djfr) [inline src/EShopWorld.Tools/Commands/AzScan/AzCosmosDbScanCommand.cs]: No need to store the key, just the connection string. From an application point of view, they shouldn't be doing anything with the key in isolation.

File: src/EShopWorld.Tools/Commands/AzScan/AzRedisScanCommand.cs [hunk 4]
```diff
@@ -21,13 +21,14 @@ protected override async Task<int> RunScanAsync(IAzure client)
...
-                if (!CheckBasicFilters(redis.Name))
+                if (!CheckRegion(redis.RegionName))
                     continue;
 
                 var name = redis.Name.Contains('-')
                     ? redis.Name.Remove(redis.Name.LastIndexOf('-')) : redis.Name;
 
-                await SetKeyVaultSecretAsync(name, redis.Keys.PrimaryKey);
+                await SetKeyVaultSecretAsync("Redis", name, "PrimaryConnectionString", $"{redis.HostName},password={redis.Keys.PrimaryKey},ssl=True,abortConnect=False");
+                await SetKeyVaultSecretAsync("Redis", name, "PrimaryKey", redis.Keys.PrimaryKey);
```
Comment (djfr) [inline src/EShopWorld.Tools/Commands/AzScan/AzRedisScanCommand.cs]: Same as with CosmosDB (and anything else that this applies to like ServiceBus)

File: src/EShopWorld.Tools/Commands/AzScan/AzScanCommandBase.cs [hunk 5]
```diff
@@ -73,25 +84,15 @@ protected virtual Task<int> RunScanAsync([NotNull] IAzure client)
...
-                return true;
-
-            var regexp = new Regex(regexpStr);
-            return regexp.IsMatch(value);
-        }
-
-        protected async Task SetKeyVaultSecretAsync(string name, string value)
-        {
-            await KeyVaultClient.SetSecretWithHttpMessagesAsync($"https://{KeyVaultName}.vault.azure.net/", name, value);
+        protected async Task SetKeyVaultSecretAsync(string prefix, string name, string suffix, string value)
```
Comment (djfr) [inline src/EShopWorld.Tools/Commands/AzScan/AzScanCommandBase.cs]: I would write this as an extension method to `KeyVaultClient`

**Pull request merged**